ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -0,0 +1,596 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.commons.ByteBufferAsyncWritableChannel;
+import com.github.ambry.commons.ByteBufferReadableStreamChannel;
+import com.github.ambry.commons.LoggingNotificationSystem;
+import com.github.ambry.commons.ResponseHandler;
+import com.github.ambry.commons.ServerErrorCode;
+import com.github.ambry.config.RouterConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.messageformat.BlobProperties;
+import com.github.ambry.network.NetworkClient;
+import com.github.ambry.network.NetworkClientErrorCode;
+import com.github.ambry.network.RequestInfo;
+import com.github.ambry.network.ResponseInfo;
+import com.github.ambry.protocol.RequestOrResponse;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+/**
+ * Tests for {@link GetBlobOperation}
+ */
+public class GetBlobOperationTest {
+  private static final int MAX_PORTS_PLAIN_TEXT = 3;
+  private static final int MAX_PORTS_SSL = 3;
+  private static final int CHECKOUT_TIMEOUT_MS = 1000;
+  private static final int BLOB_USER_METADATA_SIZE = 10;
+
+  private final MockTime time = new MockTime();
+  private final Map<Integer, GetOperation> correlationIdToGetOperation = new HashMap<>();
+  private final Random random = new Random();
+  private final FutureResult<ReadableStreamChannel> operationFuture = new FutureResult<>();
+
+  private int requestParallelism = 2;
+  private int successTarget = 1;
+  private int maxChunkSize;
+  private int blobSize = 1000;
+  private RouterConfig routerConfig;
```

Comment:
the idea is to construct the router after setting config variables custom to the tests. So the router and the properties have to be constructed outside of the router. Since `doPut()` is the first thing each test does, we are constructing all these there.